stylix/palette.nix

```diff
@@ -94,11 +129,61 @@ in
         description = "The palette generated by the palette generator.";
         readOnly = true;
         internal = true;
-        default = (lib.importJSON cfg.generated.json) // {
-          author = "Stylix";
-          scheme = "Stylix";
-          slug = "stylix";
-        };
+        default =
+          let
+            colors = builtins.mapAttrs (
+              _: value:
+              lib.concatMapStrings
+                (
+                  component:
+                  lib.strings.fixedWidthString 2 "0" (
+                    lib.toHexString (lib.strings.toInt component)
+                  )
+                )
+                (
+                  builtins.match ''rgb\(([[:digit:]]+)\.0, ([[:digit:]]+)\.0, ([[:digit:]]+)\.0\)'' value
+                )
+            ) (lib.importJSON cfg.generated.json).colors.${cfg.themeGeneration.polarity};
+          in
+          with colors;
+          if cfg.themeGeneration.polarity == "light" then
+            {
+              base00 = background;
+              base01 = surface_container;
+              base02 = surface_container_highest;
+              base03 = outline;
+              base04 = on_surface_variant;
+              base05 = on_surface;
+              base06 = on_secondary_fixed;
+              base07 = on_primary_container;
+              base08 = error;
+              base09 = on_tertiary;
+              base0A = on_secondary_container;
+              base0B = on_secondary_fixed_variant;
+              base0C = on_primary_fixed;
+              base0D = surface_tint;
+              base0E = on_tertiary_fixed;
+              base0F = on_error_container;
```

Comment:
This seems like a significant change to stylix's API. Have you considered how to provide aliases for the old names?

Doing so may also make this PR easier to review; you could have this PR add silent aliases _without_ any treewide change, then introducing warnings and a treewide migration in a followup PR.

Reply:
the keys were the same before the update

Reply:
Ah, I see. It is only the generated JSON that has different keys, the `pallet` option's value is unchanged AFAICT.

I must've been misreading it before.